most-adapter/src/index.ts

```diff
@@ -3,39 +3,78 @@ import {
   Observer,
   StreamSubscribe,
   DisposeFunction,
-  Subject,
+  Subject as CycleSubject,
 } from '@cycle/base';
 
-import {Stream} from 'most';
-import {subject} from 'most-subject';
+import {Stream, Sink, Disposable, Scheduler} from 'most';
+import {
+  BasicSubjectSource,
+  Subject as MostSubject,
+  subject as getMostSubject
+} from 'most-subject';
 import hold from '@most/hold';
 
+class AdaptedSubjectDisposable<T> implements Disposable<T> {
+  private disposed: boolean = false;
+  constructor (private subjectDisposable: Disposable<T>, private adaptedDisposer: DisposeFunction) {}
+
+  dispose () {
+    if (!this.disposed) {
+      this.disposed = true;
+      this.adaptedDisposer();
+      return this.subjectDisposable.dispose();
+    }
+  }
+}
+
+class AdaptedSubjectSource<T> extends BasicSubjectSource <T> {
+  constructor(protected originStream: any, protected originStreamSubscribe: StreamSubscribe) {
+    super();
+  }
+
+  run(sink: Sink<T>, scheduler: Scheduler): Disposable<T> {
+    const that = this;
+    let _dispose: DisposeFunction | void;
+    const _subjectDisposable = super.run(sink, scheduler);
+
+    setTimeout(() => {
+      _dispose = that.originStreamSubscribe(that.originStream, {
+        next: (x: T) => that.next(x),
+        error: (err: Error) => that.error(err),
+        complete: (x?: T) => {
+          that.complete(x);
+          if (typeof _dispose === 'function') {
+            _dispose();
+          }
+
+          _subjectDisposable.dispose();
+        }
+      });
+    }, 4);
```

Comment:
http://stackoverflow.com/questions/9647215/what-is-minimum-millisecond-value-of-settimeout

It's just a habit I have after reading about the 4ms minimum which I just now read is in W3C but not in WHATWG.